### PR TITLE
[SPARK-33897][SQL]Can't set option 'cross' in join method.

### PR DESCRIPTION
**Pull request by kozakana**
### What changes were proposed in this pull request?

[The PySpark documentation](https://spark.apache.org/docs/3.0.1/api/python/pyspark.sql.html#pyspark.sql.DataFrame.join) says "Must be one of: inner, cross, outer, full, fullouter, full_outer, left, leftouter, left_outer, right, rightouter, right_outer, semi, leftsemi, left_semi, anti, leftanti and left_anti."
However, I get the following error when I set the cross option.

```
scala> val df1 = spark.createDataFrame(Seq((1,"a"),(2,"b")))
df1: org.apache.spark.sql.DataFrame = [_1: int, _2: string]

scala> val df2 = spark.createDataFrame(Seq((1,"A"),(2,"B"), (3, "C")))
df2: org.apache.spark.sql.DataFrame = [_1: int, _2: string]

scala> df1.join(right = df2, usingColumns = Seq("_1"), joinType = "cross").show()
java.lang.IllegalArgumentException: requirement failed: Unsupported using join type Cross
  at scala.Predef$.require(Predef.scala:281)
  at org.apache.spark.sql.catalyst.plans.UsingJoin.<init>(joinTypes.scala:106)
  at org.apache.spark.sql.Dataset.join(Dataset.scala:1025)
  ... 53 elided
```

### Why are the changes needed?

The documentation says cross option can be set, but when I try to set it, I get an java.lang.IllegalArgumentException.

### Does this PR introduce _any_ user-facing change?

Accepting this PR fix will behave the same as the documentation.

### How was this patch tested?

There is already a test for [JoinTypes](https://github.com/apache/spark/blob/1b9fd67904671ea08526bfb7a97d694815d47665/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/JoinTypesTest.scala), but I can't find a test for the join option itself.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala [hunk 1]
```diff
@@ -102,7 +102,7 @@ case class NaturalJoin(tpe: JoinType) extends JoinType {
 }
 
 case class UsingJoin(tpe: JoinType, usingColumns: Seq[String]) extends JoinType {
-  require(Seq(Inner, LeftOuter, LeftSemi, RightOuter, FullOuter, LeftAnti).contains(tpe),
+  require(Seq(Inner, LeftOuter, LeftSemi, RightOuter, FullOuter, LeftAnti, Cross).contains(tpe),
```
Comment (HyukjinKwon) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala:105]: Hm, I think you should use `DataFrame.crossJoin` explicitly.

Comment (kozakana) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala:105]: Does that mean it's better to modify the documentation?

[The join document](https://spark.apache.org/docs/3.0.1/api/python/pyspark.sql.html#pyspark.sql.DataFrame.join) says:
```
how – str, default inner. Must be one of: inner, cross, outer, full, fullouter, full_outer, left, leftouter, left_outer, right, rightouter, right_outer, semi, leftsemi, left_semi, anti, leftanti and left_anti.
```
It says that cross is configurable.

**Comment by HyukjinKwon** [comment]
ok to test

**Comment by HyukjinKwon** [comment]
Okay seems we should fix. @kozakana can you add a test case?

**Comment by HyukjinKwon** [comment]
cc @cloud-fan and @hvanhovell do you remember if we intentionally disabled cross join in `join` APIs? It's documented that `cross` is supported in `join` API. We should either fix the documentation or the codes.

**Comment by cloud-fan** [comment]
It was probably a mistake. Can we add a test?

**Comment by kozakana** [comment]
@HyukjinKwon
Thank you for confirming. I try add a test case.

**Comment by kozakana** [comment]
@HyukjinKwon

`join` methods has
```
def join(right: Dataset[_], joinExprs: Column, joinType: String): DataFrame
def join(right: Dataset[_], joinExprs: Column): DataFrame
def join(right: Dataset[_], usingColumns: Seq[String], joinType: String): DataFrame
def join(right: Dataset[_], usingColumns: Seq[String]): DataFrame
def join(right: Dataset[_], usingColumn: String): DataFrame
def join(right: Dataset[_]): DataFrame
```
[ref. org.apache.spark.sql.Dataset](https://spark.apache.org/docs/3.0.1/api/scala/org/apache/spark/sql/Dataset.html)

`join` method always requires a column name. So it may have been implemented as `crossJoin`. And, I discovery [this comment](https://github.com/apache/spark/blob/8197ee3b15265d39f05f192934b7d7e661713eaa/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L993-L995).
https://github.com/apache/spark/blob/8197ee3b15265d39f05f192934b7d7e661713eaa/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L993-L995

I suggest remove the cross option from the join method and use only the `crossJoin` method. What do you think?

**Comment by cloud-fan** [comment]
@kozakana do you mean the current behavior already makes sense?

**Comment by HyukjinKwon** [comment]
@kozakana, I think you can specify empty column names in:

```scala
def join(right: Dataset[_], usingColumns: Seq[String], joinType: String): DataFrame
def join(right: Dataset[_], usingColumns: Seq[String]): DataFrame
```

e.g.)

```scala
join(right, Nil)
```

**Comment by kozakana** [comment]
Thanks! I try it.

**Comment by kozakana** [comment]
@HyukjinKwon
Thanks for your advice. I added test of cross join in [org.apache.spark.sql.DataFrameJoinSuite](https://github.com/apache/spark/pull/30803/commits/690ef81da1b06c189201782fe4cb1c332dd74287).

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
@kozakana, can you file a JIRA as guided in http://spark.apache.org/contributing.html? SPARK-30803 seems an unrelated JIRA.

**Comment by kozakana** [comment]
ok! I understand the number and I can post the JIRA.

**Comment by kozakana** [comment]
@HyukjinKwon 
I created [SPARK-33897](https://issues.apache.org/jira/browse/SPARK-33897) in JIRA and modify title of github issue.

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.1.

**Pull request closed without merging**